homeassistant/components/myio/sensor.py

```diff
@@ -0,0 +1,112 @@
+    if _server_status.startswith("Online"):
+        try:
+            for _sensor in _sensors:
+                _actual_sensor = _sensors[_sensor]
+                _sensor_id = _actual_sensor["id"]
+                if (
+                    0 < _sensor_id <= 100
+                    or (100 < _sensor_id < 200 and _actual_sensor["hum"] != 0)
+                    or (_sensor_id == 200)
+                ):
+                    sensor_entities.append(
+                        MyIOSensor(hass, _sensor_id, _server_data, _server_name)
+                    )
+        except (ValueError, Exception):  # pylint: disable=broad-except
+            _LOGGER.debug("%s - No sensor data", _server_name)
+
+        async_add_entities(sensor_entities)
+
+
+class MyIOSensor(Entity):
+    """Implementation of a myIO-Server."""
+
+    def __init__(self, hass, _sensor_id, _server_data, _server_name):
+        """Initialize the sensor."""
+        self._sensor_id = _sensor_id
+        self._server_name = _server_name
+        self._server_data = _server_data
+        self._sensors = _server_data["sensors"]
+        self._number = 0
+        self._state = 0
+
+        # initialize the correct number in the sensors list of the sensor
+        for _sensor in self._sensors:
+            if self._sensors[_sensor]["id"] == _sensor_id:
+                self._number = _sensor
+                break
+        self.entity_id = f"sensor.{_server_name}_{self._sensor_id}"
+        self._name = self._sensors[str(self._number)]["description"]
+        # initialize if the sensor is thermo-, humidity-, or imp counter"""
+        if int(self._number) < 100:
+            self._state = self._sensors[str(self._number)]["temp"] / 100
+        elif int(self._number) < 200:
+            self._state = self._sensors[str(self._number)]["hum"] / 10
+        else:
+            self._state = self._sensors[str(self._number)]["imp"]
+
+    @property
+    def scan_interval(self):
+        """Return the unique id."""
+        return SCAN_INTERVAL
+
+    @property
+    def unique_id(self):
+        """Return unique ID for light."""
+        return f"server name = {self._server_name}, id = {self._sensor_id}"
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        if int(self._number) >= 200:
+            return "Imp.Counter"
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        string_to_send = ""
+        if int(self._number) < 100:
+            string_to_send = TEMP_CELSIUS
+        elif int(self._number) < 200:
+            string_to_send = "%"
+        else:
+            string_to_send = self._name
+        return string_to_send
+
+    def update(self):
+        """Fetch new state data for the sensor."""
+
+        self._server_data = self.hass.data[self._server_name]
```

Comment:
Instead of this logic, one can subscribe the entity itself with the DataUpdateCoordinator.
In that case, the entity doesn't have to poll for data/updates itself.

See: <https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities>

Reply:
The entity cannot poll anything... it just reads the values from hass.data database... The poll is automatic.. the whole database updating continously with the __init__.py -s coordinator... this is the only place, when the myIO server is polled...
I tried to use here, the DataUpdateCoordinator, but I wasn't able to make  it work :( 
The method is that... in the background the the database is updating (refresh interval), and all of the enities are refreshing themself in another intervall from the database (hass.data...)